aptos-move/move-examples/feature-sandbox/sources/sandbox_messaging.move

```diff
@@ -0,0 +1,213 @@
+    ///
+    /// Additionally, here's an example of returning a tuple.  Multiple values can be returned at once.
+    public(friend) fun get_message_and_revision(address: address): (u64, String) acquires MessageHolder {
+        assert!(exists<MessageHolder<MessageChangeEvent>>(address), not_found(ENO_MESSAGE_HOLDER));
+        let holder = borrow_global<MessageHolder<MessageChangeEvent>>(address);
+        (event::counter(&holder.message_change_events), holder.message)
+    }
+
+    #[view]
+    /// Checks if the admin account is the given address
+    ///
+    /// Private functions (without public or friend), can only be called within the same module.
+    ///
+    /// View functions can also apply to private functions.  View functions must return a value.
+    fun check_if_admin(address: address): bool {
+        assert!(address == @feature_sandbox, permission_denied(ENOT_ADMIN));
+        true
+    }
+
+    /// An admin private entry function for setting messages of accounts that have already created message holders
+    ///
+    /// Private entry functions allow you to create externally callable functions in a transaction that cannot
+    /// be called from another module.  Entry functions cannot take structs as arguments, and cannot return values.
+    ///
+    /// This allows only the admin (the deployer) to override messages in the account
+    entry fun set_message_admin(
+        account: signer,
+        message_address: address,
+        message: String
+    ) acquires MessageHolder {
+        let account_addr = signer::address_of(&account);
+        check_if_admin(account_addr);
+        assert!(exists<MessageHolder<MessageChangeEvent>>(message_address), not_found(ENO_MESSAGE_HOLDER));
+        set_message_inline(message_address, message);
+    }
+
+    /// Sets a message for the signer's account
+    ///
+    /// Public entry functions allow you to create externally callable functions that can be called from other modules.
+    /// Entry functions cannot take structs as arguments, and cannot return values.
+    ///
+    /// Only the owner of the account holding the resource can update the message.
+    public entry fun set_message(account: signer, message: String)
+    acquires MessageHolder {
+        let account_addr = signer::address_of(&account);
+
+        // Create the message holder if it doesn't exist yet
+        if (!exists<MessageHolder<MessageChangeEvent>>(account_addr)) {
+            move_to(&account, MessageHolder {
+                message,
+                message_change_events: account::new_event_handle<MessageChangeEvent>(&account),
+            })
+        } else {
+            // Otherwise just update the existing message
+            set_message_inline(account_addr, message);
+        }
+    }
+
+    /// Updates an existing message in the message holder
+    ///
+    /// An inline function is inlined into the
+    inline fun set_message_inline(address: address, message: String) acquires MessageHolder {
```

Comment:
Although inline is allowed here, what's the purpose of making this inline?

Reply:
Purpose is this example has every feature jammed into one, so I can run the compiler & other tests directly against it.  It is pretty pointless to have it inlined.

Reply:
Alright